docs/features/traffic-management/smi.md

```diff
@@ -0,0 +1,58 @@
+# Service Mesh Interface (SMI)
+
+!!! important
+    This is not implemented yet and is scheduled to be part of the v0.9 release
+
+[Service Mesh Interface](https://smi-spec.io/) (SMI) is a standard interface for service meshes on Kubernetes leveraged by many Service Mesh implementations (like Linkerd). SMI offers this functionality through a set of CRDs, and the Argo Rollouts controller creates these resources to manipulate the traffic routing into the desired state. 
+
+The Argo Rollout controller achieves traffic shaping by creating and manipulating the [TrafficSplit CR](https://github.com/servicemeshinterface/smi-spec/blob/master/traffic-split.md). A TrafficSplit describes the desired traffic routing for an application and relies on the underlying Service Meshes implement that desired state. Instead of worrying about the details of a specific service mesh, a user needs to specify a root Service that clients use to communicate and a list of backends consisting of a Service and weight. The Service Mesh implementing SMI uses this spec to route traffic to the backends Services based on the weights of the backends. For Rollout users, the Argo Rollout controller create and manipulates the TrafficSplit using the following information:
+
+- Canary Service: Name of the service that sends traffic only to the canary pods
+- Stable Service: Name of the service that sends traffic only to the stable po ds
+- Root Service: Name of the service that clients use to communicate. If a request comes to this root service not through a proxy, the standard Kubernetes service routing will be used.
+
+Below is an example of a Rollout with all the required fields configured:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: rollout-example
+spec:
+  ...
+  strategy:
+    canary:
+      steps:
+      - setWeight: 5
+      - pause:
+          duration: 600
+      canaryService: canary-svc # required
+      stableService: stable-svc # required
+      trafficRouting:
+        smi:
+         rootService: root-svc # required
```

Comment:
Any reason to expose the `root-svc` to users? You should be able to treat `stable-svc` the same way, and then this all works for clients that don't have traffic split support (eg. anything unmeshed).

Reply:
Are you suggesting something like the following?
```yaml
apiVersion: split.smi-spec.io/v1alpha1
kind: TrafficSplit
metadata:
  name: rollout-example-traffic-split
spec:
  service: stable-svc
  backends:
  - service: stable-svc
    weight: 95
  - service: canary-svc
    weight: 5
```
Wouldn't that violate the fourth tradeoff listed in the [SMI TrafficSplit spec](https://github.com/servicemeshinterface/smi-spec/blob/master/traffic-split.md#tradeoffs):
```
* TrafficSplits cannot be self-referential
```

If having a self-referential TrafficSplit is allowed, then the rootService could be optional and it should reference the stableService if anything.

Reply:
> * TrafficSplits cannot be self-referential

This is specifically related to recursion, you can't have a traffic split that forwards to a traffic split. In retrospect, I probably could have worded that a little bit better.

> If having a self-referential TrafficSplit is allowed, then the rootService could be optional and it should reference the stableService if anything.

I think you're right. I'll need to double check out Flagger's doing it, but I believe that's the way they're making it all work under the covers.

Reply:
Great! Just pushed the change to make the rootService optional